src/Build/Definition/ProjectProperty.cs

```diff
@@ -110,6 +110,27 @@ string IProperty.EvaluatedValueEscaped
             }
         }
 
+        /// <summary>
+        /// Gets or sets object's location in xml file.
+        /// </summary>
+        public (string File, int Line, int Column) Location { get; set; }
```

Comment:
Does this need to be public?

Reply:
And is it used?

Single property can have multiple locations - so this is probably last on evaluated... but given the nature of this behavior - is that a usefull information. The info about initial property assignment location (https://github.com/dotnet/msbuild/issues/2711) and reassignemnets locations should probably be emitted/stored via separate events/data

Reply:
It will be used for other rules after extending https://github.com/dotnet/msbuild/blob/066afd44ea4d23b5c4649acc001b118e5cddbd1d/src/Build/BuildCheck/Infrastructure/BuildEventsProcessor.cs#L54

(we were discussing it offline)

Reply:
it has to be public :
Error (active)	CS0737	'ProjectProperty' does not implement interface member 'IProperty.Location'. 'ProjectProperty.Location' cannot implement an interface member because it is not public.